server-spi/src/main/java/org/keycloak/credential/CredentialMetadata.java

```diff
@@ -0,0 +1,63 @@
+package org.keycloak.credential;
+
+public class CredentialMetadata {
+    LocalizedMessage infoMessage;
+    LocalizedMessage warningMessageTitle;
+    LocalizedMessage warningMessageDescription;
+    CredentialModel credentialModel;
+
+    public CredentialModel getCredentialModel() {
+        return credentialModel;
+    }
+
+    public void setCredentialModel(CredentialModel credentialModel) {
+        this.credentialModel = credentialModel;
+    }
+
+    public LocalizedMessage getInfoMessage() {
+        return infoMessage;
+    }
+
+    public LocalizedMessage getWarningMessageTitle() {
+        return warningMessageTitle;
+    }
+
+    public LocalizedMessage getWarningMessageDescription() {
+        return warningMessageDescription;
+    }
+
+    public void setWarningMessageTitle(String key, String... parameters) {
+        LocalizedMessage message = new LocalizedMessage();
+        message.key = key;
+        message.parameters = parameters;
+        this.warningMessageTitle = message;
+    }
+
+    public void setWarningMessageDescription(String key, String... parameters) {
+        LocalizedMessage message = new LocalizedMessage();
+        message.key = key;
+        message.parameters = parameters;
+        this.warningMessageDescription = message;
+    }
+
+    public void setInfoMessage(String key, String... parameters) {
+        LocalizedMessage message = new LocalizedMessage();
+        message.key = key;
+        message.parameters = parameters;
+        this.infoMessage = message;
+    }
+
+    class LocalizedMessage {
```

Comment:
If this type is exposed via Public API, e.g. #getInfoMessage(), then the type should be `public static`

Reply:
OK.